File: Core kernel Yaiwes/Backend watchdog workflow adaptativo/sandbox-runtime/iii-sandbox/packages/sdk-rust/tests/env_test.rs
use std::collections::HashMap;
use std::sync::Arc;

use iii_sandbox::env::EnvManager;
use iii_sandbox::{ClientConfig, HttpClient};

fn make_env(url: &str) -> EnvManager {
    let client = Arc::new(HttpClient::new(ClientConfig {
        base_url: url.to_string(),
        token: None,
        timeout_ms: None,
    }).unwrap());
    EnvManager::new(client, "sbx-1".into())
}

#[tokio::test]
async fn test_get_returns_value() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/env/get")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(
            serde_json::json!({
                "key": "API_KEY",
                "value": "secret123",
                "exists": true
            })
            .to_string(),
        )
        .create_async()
        .await;

    let env = make_env(&server.url());
    let result = env.get("API_KEY").await.unwrap();
    assert_eq!(result.key, "API_KEY");
    assert_eq!(result.value, Some("secret123".to_string()));
    assert!(result.exists);
    mock.assert_async().await;
}

#[tokio::test]
async fn test_get_returns_null_for_missing_key() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/env/get")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(
            serde_json::json!({
                "key": "MISSING",
                "value": null,
                "exists": false
            })
            .to_string(),
        )
        .create_async()
        .await;

    let env = make_env(&server.url());
    let result = env.get("MISSING").await.unwrap();
    assert_eq!(result.key, "MISSING");
    assert!(result.value.is_none());
    assert!(!result.exists);
    mock.assert_async().await;
}

#[tokio::test]
async fn test_set_with_multiple_vars() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/env")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(
            serde_json::json!({
                "set": ["FOO", "BAR"],
                "count": 2
            })
            .to_string(),
        )
        .create_async()
        .await;

    let env = make_env(&server.url());
    let mut vars = HashMap::new();
    vars.insert("FOO".into(), "1".into());
    vars.insert("BAR".into(), "2".into());
    let result = env.set(vars).await.unwrap();
    assert_eq!(result.count, 2);
    assert_eq!(result.set.len(), 2);
    mock.assert_async().await;
}

#[tokio::test]
async fn test_list() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("GET", "/sandbox/sandboxes/sbx-1/env")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(
            serde_json::json!({
                "vars": {"PATH": "/usr/bin", "HOME": "/root"},
                "count": 2
            })
            .to_string(),
        )
        .create_async()
        .await;

    let env = make_env(&server.url());
    let result = env.list().await.unwrap();
    assert_eq!(result.count, 2);
    assert_eq!(result.vars.get("PATH").unwrap(), "/usr/bin");
    assert_eq!(result.vars.get("HOME").unwrap(), "/root");
    mock.assert_async().await;
}

#[tokio::test]
async fn test_delete() {
    let mut server = mockito::Server::new_async().await;
    let mock = server
        .mock("POST", "/sandbox/sandboxes/sbx-1/env/delete")
        .with_status(200)
        .with_header("content-type", "application/json")
        .with_body(serde_json::json!({"deleted": "OLD_VAR"}).to_string())
        .create_async()
        .await;

    let env = make_env(&server.url());
    let result = env.delete("OLD_VAR").await.unwrap();
    assert_eq!(result.deleted, "OLD_VAR");
    mock.assert_async().await;
}

File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/test/sdk/env.test.ts
import { describe, it, expect, vi, beforeEach } from "vitest"
import { EnvManager } from "../../packages/sdk/src/env.js"
import { HttpClient } from "../../packages/sdk/src/client.js"

describe("EnvManager", () => {
  let mockClient: HttpClient
  let env: EnvManager

  beforeEach(() => {
    mockClient = {
      get: vi.fn(),
      post: vi.fn(),
      del: vi.fn(),
      stream: vi.fn(),
    } as unknown as HttpClient
    env = new EnvManager(mockClient, "sbx_test")
  })

  describe("get", () => {
    it("calls correct endpoint with key", async () => {
      ;(mockClient.post as any).mockResolvedValue({ key: "FOO", value: "bar", exists: true })
      const result = await env.get("FOO")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/env/get",
        { key: "FOO" },
      )
      expect(result).toEqual({ key: "FOO", value: "bar", exists: true })
    })

    it("returns exists=false for missing variable", async () => {
      ;(mockClient.post as any).mockResolvedValue({ key: "MISSING", value: null, exists: false })
      const result = await env.get("MISSING")

      expect(result.exists).toBe(false)
      expect(result.value).toBeNull()
    })
  })

  describe("set", () => {
    it("calls correct endpoint with vars", async () => {
      ;(mockClient.post as any).mockResolvedValue({ set: ["A", "B"], count: 2 })
      const result = await env.set({ A: "1", B: "2" })

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/env",
        { vars: { A: "1", B: "2" } },
      )
      expect(result).toEqual({ set: ["A", "B"], count: 2 })
    })
  })

  describe("list", () => {
    it("calls correct endpoint", async () => {
      const mockVars = { HOME: "/root", PATH: "/usr/bin" }
      ;(mockClient.get as any).mockResolvedValue({ vars: mockVars, count: 2 })
      const result = await env.list()

      expect(mockClient.get).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/env",
      )
      expect(result.vars).toEqual(mockVars)
      expect(result.count).toBe(2)
    })

    it("returns empty vars when sandbox has none", async () => {
      ;(mockClient.get as any).mockResolvedValue({ vars: {}, count: 0 })
      const result = await env.list()

      expect(result.vars).toEqual({})
      expect(result.count).toBe(0)
    })
  })

  describe("delete", () => {
    it("calls correct endpoint with key", async () => {
      ;(mockClient.post as any).mockResolvedValue({ deleted: "OLD_VAR" })
      const result = await env.delete("OLD_VAR")

      expect(mockClient.post).toHaveBeenCalledWith(
        "/sandbox/sandboxes/sbx_test/env/delete",
        { key: "OLD_VAR" },
      )
      expect(result).toEqual({ deleted: "OLD_VAR" })
    })
  })
})
